src/Network/Wai/Auth/AppRoot.hs

```diff
@@ -0,0 +1,61 @@
+{-# LANGUAGE OverloadedStrings #-}
+module Network.Wai.Auth.AppRoot
+  ( smartAppRoot
```

Comment:
This was a copy-pasted from here: https://github.com/fpco/wai-middleware-crowd/blob/master/src/Network/Wai/Approot.hs#L34
Which I further simplified a bit, in particular I removed reading `APPROOT` environment variable, since that can be achieved while reading the yaml config file. It seems like the link above is vastly different in that it doesn't check `https` related headers, that's why you probably called it `smart` :)